frontend/ui/font.lua

```diff
@@ -144,12 +188,31 @@ function Font:getFace(font, size)
         -- @field hash hash key for this font face
         face_obj = {
             orig_font = font,
+            realname = realname,
             size = size,
             orig_size = orig_size,
             ftface = face,
-            hash = hash
+            hash = hash,
         }
         self.faces[hash] = face_obj
+
+        -- Callback to be used by libkoreader-xtext.so to get Freetype
+        -- instantiated fallback fonts when needed for shaping text
+        face_obj.getFallbackFont = function(num)
+            return _getFallbackFont(face_obj, num)
+        end
+        -- Font features, to be given by libkoreader-xtext.so to HarfBuzz.
+        -- (Could be tweaked by font if needed. Note that NotoSans does not
+        -- have common ligatures, like for "fi" or "fl", so we won't see
+        -- them in the UI.)
+        face_obj.hb_features = { "+kern", "+liga" }
```

Comment:
So is this technically redundant with HB defaults? I'm slightly confused by the comment following this line.

Reply:
Half :) I've indeed been unclear, but that's mostly just cut and pasted from crengine.
And what the HB defaults are are also hard to track. It's in its `src/hb-ot-shape.cc`.
So, it looks to me `+liga` is the default, but not `+kern` (or may be it is...).
I'm just setting here the defaults we use in crengine HB full/best, even if one of them is already the default.
The commented next line is the list of all stuff mentionned in HB "good", that I just cut and pasted with all disabled to see if enabling +liga / +kern made a difference :)
I don't plan on being the reference as I'm not sure what it is :|. I'll update the comment to not state anything about "default" but "If we'd wanted to disable all the features HB may enable" :)